### PR TITLE
Core Content API service cleanup 

### PR DESCRIPTION
**Pull request by Convly**
### What does it do?

- Move core content-API's permission logic into its own file.
- Add precise JSDoc with types and explanations

### Why is it needed?

Make the code more readable and maintainable

### How to test it?

Purely architectural changes, nothing should have changed in terms of behavior.

### CODE REVIEW

File: packages/core/strapi/lib/services/content-api/index.js [hunk 1]
```diff
@@ -1,114 +1,13 @@
...
-
-    // If the action isn't registered into the action provider, then ignore the permission and warn the user
-    if (!action) {
-      strapi.log.debug(
-        `Unknown action "${permission.action}" supplied when registering a new permission`
-      );
-      return false;
-    }
-  };
+const instanciatePermissionsUtilities = require('./permissions');
```
Comment (innerdvations) [inline packages/core/strapi/lib/services/content-api/index.js:3]: instantiate :)

Comment (Convly) [reply packages/core/strapi/lib/services/content-api/index.js]: oops, thanks!

**Comment by Convly** [comment]
Note: unit back failing because of test that is fixed by https://github.com/strapi/strapi/pull/14268

**Review by Bassel17 — APPROVED** [review]
LGTM

**Review by innerdvations — APPROVED** [review]

**Comment by Convly** [comment]
Waiting for #14268 to be merged to merge this one

**Pull request merged**